collector/collector.go

```diff
@@ -119,6 +123,11 @@ func (c *Collector) Collect() (*Result, error) {
 
 		res, err := httpClient.Do(req)
 		if err != nil {
+			// Authorization error signals token is invalid
+			if res.StatusCode == 401 {
+				log.Printf("DEBUG HTTP 401 returned from uri=%s\n", req.URL)
+				return nil, fmt.Errorf("http 401 response received %w", ErrUnauthorized)
+			}
```

Comment:
If the http request gets a well-formed response from the server, `httpClient.Do` will return a `nil` error. A 401 is still a well-formed response. So you have to check the status code outside the `if err != nil {`

See https://pkg.go.dev/net/http#Client.Do: "A non-2xx status code doesn't cause an error."